### PR TITLE
feat [IDP-1884] :  Added the release notes for v0.17.0

### PR DESCRIPTION
**Pull request by Debanitrkl**
# Harness Developer Pull Request
Added the release notes for v0.17.0

## What Type of PR is This?

- [ ] Issue
- [x] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [x] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

File: release-notes/internal-developer-portal.md [hunk 1]
```diff
@@ -20,6 +20,31 @@ Review the notes below for details about recent changes to Harness Internal Deve
 
 ## December 2023
 
+### Version 0.17.0
+
+<!-- Dec 21, 2023 -->
```
Comment (OrkoHunter) [inline release-notes/internal-developer-portal.md:25]: IMO we should share the release date as well with the version.

File: release-notes/internal-developer-portal.md [hunk 2]
```diff
@@ -20,6 +20,31 @@ Review the notes below for details about recent changes to Harness Internal Deve
...
+
+In this release, we're excited to unveil features like the HTTP actions support in the scaffolder-backend 🌐, the innovative Developer Portal stage for seamless self-service flows 🚀, and the integration of OPA/Governance policies to ensure streamlined and secure management 🛡️. Plus, we've squashed some pesky bugs 🐛 to make your experience smoother than ever!
+
+- **Docs:** [Supported Custom Actions](https://developer.harness.io/docs/internal-developer-portal/flows/custom-actions), [OPA](Link to be added), [IDP Stage](https://6582ce9b1eb6070097c8b473--harness-developer.netlify.app/docs/internal-developer-portal/flows/idp-stage)
+
+#### New features and enhancements
+
+- Added support for http actions in the scaffolder-backend.  [IDP-1853].
```
Comment (OrkoHunter) [inline release-notes/internal-developer-portal.md:35]: Link to https://roadie.io/backstage/plugins/scaffolder-http-requests/

File: release-notes/internal-developer-portal.md [hunk 3]
```diff
@@ -20,6 +20,31 @@ Review the notes below for details about recent changes to Harness Internal Deve
...
+
+In this release, we're excited to unveil features like the HTTP actions support in the scaffolder-backend 🌐, the innovative Developer Portal stage for seamless self-service flows 🚀, and the integration of OPA/Governance policies to ensure streamlined and secure management 🛡️. Plus, we've squashed some pesky bugs 🐛 to make your experience smoother than ever!
+
+- **Docs:** [Supported Custom Actions](https://developer.harness.io/docs/internal-developer-portal/flows/custom-actions), [OPA](Link to be added), [IDP Stage](https://6582ce9b1eb6070097c8b473--harness-developer.netlify.app/docs/internal-developer-portal/flows/idp-stage)
+
+#### New features and enhancements
+
+- Added support for http actions in the scaffolder-backend.  [IDP-1853]. 
+- Added a new stage, specifically for IDP, knows as Developer Portal to help with the self service flows (presently this feature is behind a Feature Flag) [IDP-1425]
```
Comment (OrkoHunter) [inline release-notes/internal-developer-portal.md:36]: Link to docs?

File: release-notes/internal-developer-portal.md [hunk 4]
```diff
@@ -20,6 +20,31 @@ Review the notes below for details about recent changes to Harness Internal Deve
...
+- Added OPA/Governance policies for Backstage entities. [IDP-1514]
+- Added a Cookiecutter step to enable use of cookiecutter based templates in the IDP stage.[IDP-1437]
+- Added a step to register software components in IDP catalog.[IDP-1438]
+- Added a step to create a repository in your git provider and push contents along with catalog-info.yaml. [IDP-1436]
```
Comment (OrkoHunter) [inline release-notes/internal-developer-portal.md:40]: Please link to docs if available.

File: release-notes/internal-developer-portal.md [hunk 5]
```diff
@@ -20,6 +20,31 @@ Review the notes below for details about recent changes to Harness Internal Deve
...
+
+#### Bug Fixes
+
+- Fixed support for special character “-”(hyphen) in the URLs.[IDP-1804]
```
Comment (OrkoHunter) [inline release-notes/internal-developer-portal.md:44]: URLs for Kubernetes plugin configuration - I believe.

File: release-notes/internal-developer-portal.md [hunk 6]
```diff
@@ -20,6 +20,31 @@ Review the notes below for details about recent changes to Harness Internal Deve
...
 
+### Version 0.17.0
+
+<!-- Dec 21, 2023 -->
+
+Jingle bells, jingle bells, jingle all the way to Harness IDP's End-of-Year Release! 🛷 As we close the chapter on a year of incredible milestones, we want to express our heartfelt thanks to our amazing customers and prospects. Your support, feedback, and enthusiasm have been the North Star 🌟 guiding us in enhancing and evolving Harness IDP. The debut of our new module and the strides we've made together are testaments to your invaluable collaboration and spirit.
+
+In this release, we're excited to unveil features like the HTTP actions support in the scaffolder-backend 🌐, the innovative Developer Portal stage for seamless self-service flows 🚀, and the integration of OPA/Governance policies to ensure streamlined and secure management 🛡️. Plus, we've squashed some pesky bugs 🐛 to make your experience smoother than ever!
+
+- **Docs:** [Supported Custom Actions](https://developer.harness.io/docs/internal-developer-portal/flows/custom-actions), [OPA](Link to be added), [IDP Stage](https://6582ce9b1eb6070097c8b473--harness-developer.netlify.app/docs/internal-developer-portal/flows/idp-stage)
```
Comment (sandeepa-kv-jois) [inline release-notes/internal-developer-portal.md:31]: IDP stage can't be used until other dependent services like pipeline and CI makes it into prod. If we are mentioning about IDP stage on release notes, perhaps we need to mention about enablement delay.

File: release-notes/internal-developer-portal.md (hunk 1 above)
Comment (Debanitrkl) [reply release-notes/internal-developer-portal.md:25]: This is to adhere with the new Release Notes format, let me check if we can change and add the dates.

File: release-notes/internal-developer-portal.md (hunk 3 above)
Comment (Debanitrkl) [reply release-notes/internal-developer-portal.md:36]: Will add it once I merge this PR. https://github.com/harness/developer-hub/pull/4667

File: release-notes/internal-developer-portal.md (hunk 4 above)
Comment (Debanitrkl) [reply release-notes/internal-developer-portal.md]: Will add it once I merge this PR. https://github.com/harness/developer-hub/pull/4667

File: release-notes/internal-developer-portal.md (hunk 5 above)
Comment (Debanitrkl) [reply release-notes/internal-developer-portal.md:52]: No it was for all the URL input for entire bunch of supported plugins not only limited to a single plugin.

File: release-notes/internal-developer-portal.md (hunk 6 above)
Comment (Debanitrkl) [reply release-notes/internal-developer-portal.md:31]: Updated it now please check.

**Review by sandeepa-kv-jois — APPROVED** [review]

**Pull request merged**